placement/algo/sharded_helper.go

```diff
@@ -230,10 +260,15 @@ func (ph *placementHelper) targetLoadForInstance(id string) int {
 }
 
 func (ph *placementHelper) moveOneShard(from, to placement.Instance) bool {
-	for _, s := range from.Shards().All() {
-		if s.State() != shard.Leaving && ph.moveShard(s, from, to) {
-			return true
-		}
+	// Prefer to move a Unknown shard first, then Initializing, then Available.
+	if ph.moveOneShardInState(from, to, shard.Unknown) {
+		return true
+	}
+	if ph.moveOneShardInState(from, to, shard.Initializing) {
+		return true
+	}
+	if ph.moveOneShardInState(from, to, shard.Available) {
+		return true
 	}
 	return false
```

Comment:
nit: take it or leave it, but could be simplified to
```
return ph.moveOneShardInState(from, to, shard.Unknown) ||
  ph.moveOneShardInState(from, to, shard.Initializing) ||
  ph.moveOneShardInState(from, to, shard.Available)

Reply:
sg